SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -882,7 +882,10 @@ - (void)sendMessageToService:(TSOutgoingMessage *)message
 
             dispatch_async([OWSDispatch sendingQueue], ^{
                 // This emulates the completion logic of an actual successful save (see below).
-                [recipient save];
+                [self.dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
+                    [message updateWithSkippedRecipient:localNumber transaction:transaction];
```

Comment:
We need to mark the local recipient as skipped for sync message when we have no linked devices. They are indeed skipped.